sql/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectSessionManager.scala

```diff
@@ -333,13 +346,7 @@ class SparkConnectSessionManager extends Logging {
   }
 
   private def newIsolatedSession(): SparkSession = {
-    val active = SparkSession.active
-    if (active.sparkContext.isStopped) {
-      assert(SparkSession.getDefaultSession.nonEmpty)
-      SparkSession.getDefaultSession.get.newSession()
-    } else {
-      active.newSession()
-    }
+    baseSession.get.newSession()
```

Comment:
This should be synchronized as well. You can also opt to make the variable volatile.